### PR TITLE
chore(trace): Add extra "data" field to the trace context

### PR DESCRIPTION
**Pull request by olksdr**
Adds extra `data` field to the trace context with nested types for better schema definition. 

#skip-changelog

### CODE REVIEW

File: relay-event-schema/src/protocol/contexts/trace.rs [hunk 1]
```diff
@@ -118,6 +118,10 @@ pub struct TraceContext {
     /// This flag only applies to events with [`Error`] type that have an associated dynamic sampling context.
     pub sampled: Annotated<bool>,
 
+    /// Arbitrary additional data on a trace.
+    #[metastructure(pii = "true")]
```
Comment (jjbayer) [inline relay-event-schema/src/protocol/contexts/trace.rs]:
```suggestion
    #[metastructure(pii = "true", bag_size = "medium")]
    #[metastructure(skip_serialization = "empty")]
```

I don't know if this is the right bag size, but I noticed other data bags have it + the `skip_serialization` attr.

**Review by jjbayer — APPROVED** [review]

File: relay-event-schema/src/protocol/contexts/trace.rs (hunk 1 above)
Comment (olksdr) [reply relay-event-schema/src/protocol/contexts/trace.rs]: @jjbayer  changed the annotations in [e2a23c3](https://github.com/getsentry/relay/pull/2899/commits/e2a23c3dc2a355701135cdd6fd749a10641cb76d)

**Review by iker-barriocanal — COMMENTED** [review]
Do we know what lives in `data`, across all SDKs? Could this result in false positives (scrubbing what shouldn't be) that degrade the product experience?

**Comment by olksdr** [comment]
@jjbayer  PTAL! 
Added explicit types for the routes in the `data` field.

File: relay-event-schema/src/protocol/contexts/trace.rs [hunk 2]
```diff
@@ -118,11 +118,58 @@ pub struct TraceContext {
...
+    pub other: Object<Value>,
+}
+
+/// The route in the application, set by React Native SDK.
+#[derive(Clone, Debug, Default, PartialEq, Empty, FromValue, IntoValue, ProcessValue)]
+#[cfg_attr(feature = "jsonschema", derive(JsonSchema))]
+pub struct Route {
+    /// Parameters assigned to this route.
+    #[metastructure(pii = "true", skip_serialization = "empty", bag_size = "medium")]
+    params: Annotated<Object<Value>>,
```
Comment (jjbayer) [inline relay-event-schema/src/protocol/contexts/trace.rs:161]: Not 100% sure if a `pii=true` child of a `pii=maybe` parent gets scrubbed automatically (I hope so). Should we add a test to `relay-pii` for an example input like `{"route": {"params": {"password": "pw-leaked"}}}`.

**Review by jjbayer — APPROVED** [review]

File: relay-event-schema/src/protocol/contexts/trace.rs (hunk 2 above)
Comment (olksdr) [reply relay-event-schema/src/protocol/contexts/trace.rs:161]: yep, added test in [a325880](https://github.com/getsentry/relay/pull/2899/commits/a325880608e6552739f796d6889a67e36a3a2a1a), seems like scrubbing is working by default as expected.

**Pull request merged**